doc/api/esm.md

```diff
@@ -1196,16 +1196,23 @@ export async function resolve(specifier, context, defaultResolve) {
 > signature may change. Do not rely on the API described below.
 
 The `getFormat` hook provides a way to define a custom method of determining how
-a URL should be interpreted. This can be one of the following:
+a URL should be interpreted. The `format` returned also affects what the
+acceptable forms of source values are for a module when parsing. This can be one
+of the following:
 
-| `format` | Description |
+| `format` | Description | Acceptable Types For `source` Returned by `getSource` or `transformSource` |
 | --- | --- |
-| `'builtin'` | Load a Node.js builtin module |
-| `'commonjs'` | Load a Node.js CommonJS module |
-| `'dynamic'` | Use a [dynamic instantiate hook][] |
-| `'json'` | Load a JSON file |
-| `'module'` | Load a standard JavaScript module (ES module) |
-| `'wasm'` | Load a WebAssembly module |
+| `'builtin'` | Load a Node.js builtin module | Not applicable |
+| `'commonjs'` | Load a Node.js CommonJS module | Not applicable |
+| `'dynamic'` | Use a [dynamic instantiate hook][] | Not applicable |
+| `'json'` | Load a JSON file | array buffer, string, or typed array |
+| `'module'` | Load a standard JavaScript module (ES module) | array buffer, string, or typed array |
+| `'wasm'` | Load a WebAssembly module | array buffer, or typed array |
+
+For text based formats like `'json'` or `'module'` if the source value is not a
+string it will be converted to a string using [`util.TextDecoder`][].
```

Comment:
Note: If the source value of a text-based format (i.e., `'json'`, `'module'`) is not a string, it will be converted to a string using [`util.TextDecoder`][].
```suggestion
Note: If the source value of a text-based format (i.e., `'json'`, `'module'`) is not a
string, it will be converted to a string using [`util.TextDecoder`][].
```